jtsync/jtsync.rb

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ENV["CURRENT_GEMFILE"] ||= File.expand_path("Gemfile", __FILE__)
+
+require "bundler"
+Bundler.setup(:default)
+require "trello"
+require "byebug"
+
+class Failed < StandardError
+end
+
+def board
+  @trello_board ||= if ENV.has_key? "TRELLO_BOARD"
+                      Trello::Board.find(ENV["TRELLO_BOARD"])
+                    else
+                      Trello::Board.find("ywSwlQpZ")
+                    end
+end
+
+
+def state_from_env
+  case ENV["JOB_STATUS"]
+  when 0 then :successful
+  else :failed
+  end
+end
+
+
+# Handling different naming conventions between opensuse/suse
+# and matrix/normal job
+# 
+# returns tuple: (list_name, job_name)
+def job_from_env
+  # is this a ci.opensuse.org matrix job? (Juno)
+  if ENV.has_key? "openstack_project"
+    return ["OpenStack", "#{ENV["JOB_NAME"]} #{ENV["openstack_project"]}"]
+  end
+
+  # is this a ci.suse.de matrix job? (Devel:Cloud:7:Staging)
+  if ENV.has_key? "project"
+    version = ENV["project"].split(":")[2]
+    raise Failed.new("Can not determine ci.suse.de matrix job name") if version.nil?
```

Comment:
@aplanas 'cannot' is pretty common from my experience. It's a bit weird, I admit :)